notebooks/Watermark photo attachments with Exif data.ipynb

```diff
@@ -0,0 +1,179 @@
+    "# and for protecting your credentials see\n",
+    "# https://developers.arcgis.com/python/guide/working-with-different-authentication-schemes/#protecting-your-credentials\n",
+    "gis = GIS(\"home\")\n",
+    "\n",
+    "# The item id of the feature layer to process attachments for\n",
+    "feature_layer_item_id = \"7a28f3f872bd4b55bfd7965bf73a46b6\"\n",
+    "# The id of the layer to use\n",
+    "feature_layer_id = 0\n",
+    "# Where clause to filter features and attachments\n",
+    "# See FeatureLayer query() for more options - https://developers.arcgis.com/python/api-reference/arcgis.features.toc.html#featurelayer\n",
+    "attachments_query = \"field_to_filter=2\"\n",
+    "\n",
+    "\n",
+    "# Converts latitude and longitude from Exif GPS info in degrees, minutes and seconds to decimal degrees\n",
+    "def convert_dms_to_dd(gps_info) -> (float, float):\n",
+    "\n",
+    "    if gps_info is None:\n",
+    "        return (0, 0)\n",
+    "\n",
+    "    lat_degrees = gps_info[ExifTags.GPS.GPSLatitude][0]\n",
+    "    lat_minutes = gps_info[ExifTags.GPS.GPSLatitude][1]\n",
+    "    lat_seconds = gps_info[ExifTags.GPS.GPSLatitude][2]\n",
+    "    lon_degrees = gps_info[ExifTags.GPS.GPSLongitude][0]\n",
+    "    lon_minutes = gps_info[ExifTags.GPS.GPSLongitude][1]\n",
+    "    lon_seconds = gps_info[ExifTags.GPS.GPSLongitude][2]\n",
+    "\n",
+    "    lat = float(lat_degrees + lat_minutes / 60 + lat_seconds / 3600)\n",
+    "    lon = float(lon_degrees + lon_minutes / 60 + lon_seconds / 3600)\n",
+    "\n",
+    "    if gps_info[ExifTags.GPS.GPSLatitudeRef] == \"S\":\n",
+    "        lat = -lat\n",
+    "    if gps_info[ExifTags.GPS.GPSLongitudeRef] == \"W\":\n",
+    "        lon = -lon\n",
+    "\n",
+    "    return (lat, lon)\n",
+    "\n",
+    "\n",
+    "# Watermarks an image with decimal degree latitude and longitude from Exif data\n",
+    "def add_dd_watermark(image_path):\n",
+    "\n",
+    "    image = Image.open(image_path)\n",
+    "    image = ImageOps.exif_transpose(image)\n",
+    "\n",
+    "    # Get the image's Exif data, convernt the lat and long to decmial degree format and formate the string for the watermark\n",
+    "    exif = image.getexif()\n",
+    "    gps_ifd = exif.get_ifd(ExifTags.IFD.GPSInfo)\n",
+    "    (lat, lon) = convert_dms_to_dd(gps_ifd)\n",
+    "    watermark_text = f\"Latitude: {lat:.4f} Longitude: {lon:.4f}\"\n",
+    "\n",
+    "    draw = ImageDraw.Draw(image)\n",
+    "\n",
+    "    # Draw the watermark in the top left corner of the image in white\n",
+    "    # For more drawing options see: https://pillow.readthedocs.io/en/stable/reference/ImageDraw.html#PIL.ImageDraw.ImageDraw.text\n",
+    "    # See https://pillow.readthedocs.io/en/stable/reference/ImageFont.html for information on how to change the font, including font size\n",
+    "    text_xy_position = (10, 10)\n",
```

Comment:
Should we consider adding a size limit for images when using this script? For instance, if the image dimensions are smaller than a 10x10 position, should we highlight this limitation? I noticed that when using a small image, part of the watermark was cut off.

Reply:
I am not sure.  We could limit it, but what we would limit the image size to?  10x10 pixels is a very small image and I would not expect anyone to be using an image that small.  We could maybe get the length of the text output using `ImageDraw.textlength`, this would give us a min to use for width of the image, but we would have to do something different for height.  There is a `ImageDraw.textbbox`, but it is only supported for TrueType fonts, which we are not using.

If we do try to limit, rather than add a size limit for the script, maybe we could print a warning message that the text maybe cutoff?

Reply:
Offline with Justin-- if we find an easy solution to this, we can add to this notebook at a later time.